### PR TITLE
mobile: removing more C wrappers

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by RyanTheOptimist — APPROVED** [review]
Boy, what a huge improvement!

File: mobile/library/cc/stream.h [hunk 1]
```diff
@@ -8,14 +8,12 @@
 #include "stream_callbacks.h"
 
 namespace Envoy {
-namespace Platform {
-
 class Engine;
-using EngineSharedPtr = std::shared_ptr<Engine>;
+namespace Platform {
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply mobile/library/cc/stream.h:12]: yeah I want there to not be 2 streams and 2 engines.

File: mobile/library/cc/stream_prototype.cc [hunk 2]
```diff
@@ -10,11 +11,10 @@ StreamPrototype::StreamPrototype(EngineSharedPtr engine) : engine_(engine) {
 }
 
 StreamSharedPtr StreamPrototype::start(bool explicit_flow_control) {
-  envoy_engine_t engine_handle = reinterpret_cast<envoy_engine_t>(engine_->engine_);
-  auto envoy_stream = init_stream(engine_handle);
-  start_stream(engine_handle, envoy_stream, callbacks_->asEnvoyHttpCallbacks(),
-               explicit_flow_control);
-  return std::make_shared<Stream>(engine_handle, envoy_stream);
+  auto envoy_stream = engine_->engine_->initStream();
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply mobile/library/cc/stream_prototype.cc:14]: going to leave it largely because I don't want to deal with CI but I'll try to be more mindful :-)

**Comment by alyssawilk** [comment]
cc @abeyad almost done with the C wrappers so also maybe want to not add more in the CONNECT PR

**Pull request merged**